src/hotspot/share/runtime/interfaceSupport.inline.hpp

```diff
@@ -201,35 +194,24 @@ class ThreadInVMfromNative : public ThreadStateTransition {
   ResetNoHandleMark __rnhm;
  public:
   ThreadInVMfromNative(JavaThread* thread) : ThreadStateTransition(thread) {
-    trans_from_native(_thread_in_vm);
+    transition_from_native(thread, _thread_in_vm);
   }
   ~ThreadInVMfromNative() {
-    assert(_thread->thread_state() == _thread_in_vm, "coming from wrong thread state");
```

Comment:
Why did you drop this assert? It guards against unexpected/unbalanced intermediate state changes.

Reply:
The assert made sense before but now we are just using transition_from_vm() which will already assert the JT is in vm.

Reply:
Ah! I  missed that.